src/demo/player.rs

```diff
@@ -110,3 +112,40 @@ fn record_player_directional_input(
         controller.intent = intent;
     }
 }
+
+#[derive(Resource, Asset, Reflect, Clone)]
+pub struct PlayerAssets {
+    #[dependency]
```

Comment:
I don't think `dependency` is an attribute a newcommer is likely to know, based on the fact that I just learned about it through this PR :D 
Would appreciate a comment explaining it